### PR TITLE
always set change record

### PR DESCRIPTION
**Pull request by akash1810**
- create sets created and lastModified
- update sets lastModified
- publish sets lastModified and published

### CODE REVIEW

File: app/model/ChangeRecord.scala [hunk 1]
```diff
@@ -3,12 +3,19 @@ package model
...
+import com.gu.pandomainauth.model.{User => PandaUser}
+import play.api.libs.json.Format
 
 case class ChangeRecord(date: DateTime, user: Option[User]) {
   def asThrift = ThriftChangeRecord(date.getMillis, user.map(_.asThrift))
 }
 
 object ChangeRecord {
-  implicit val changeRecordFormat = Jsonx.formatCaseClassUseDefaults[ChangeRecord]
+  implicit val changeRecordFormat: Format[ChangeRecord] = Jsonx.formatCaseClassUseDefaults[ChangeRecord]
```
Comment (akash1810) [inline app/model/ChangeRecord.scala]: added type because intellij said to...

File: app/model/User.scala [hunk 2]
```diff
@@ -2,12 +2,13 @@ package model
...
 import org.cvogt.play.json.Jsonx
+import play.api.libs.json.Format
 
 case class User(email: String, firstName: Option[String], lastName: Option[String]) {
   def asThrift = ThriftUser(email, firstName, lastName)
 }
 
 object User {
-  implicit val userFormat = Jsonx.formatCaseClassUseDefaults[User]
+  implicit val userFormat: Format[User] = Jsonx.formatCaseClassUseDefaults[User]
```
Comment (akash1810) [inline app/model/User.scala]: added type because intellij said to...

File: app/model/commands/CreateAtomCommand.scala [hunk 3]
```diff
@@ -32,7 +33,7 @@ case class CreateAtomCommandData(
 )
 
 object CreateAtomCommandData {
-  implicit val createAtomCommandFormat = Jsonx.formatCaseClass[CreateAtomCommandData]
+  implicit val createAtomCommandFormat: Format[CreateAtomCommandData] = Jsonx.formatCaseClass[CreateAtomCommandData]
```
Comment (akash1810) [inline app/model/commands/CreateAtomCommand.scala]: added type because intellij said to...

**Comment by akash1810** [comment]
Sadly, atoms created before this goes out will not have a `created` date. We could backfill these dates from CAPI, but not sure its worth the effort atm - will see if Editorial notice first before doing anything for now, but definitely one for sanity...

**Comment by mbarton** [comment]
Do those ones appear at the end of the list?

**Comment by akash1810** [comment]
yep

**Pull request merged**